### PR TITLE
Multitenancy

### PR DESCRIPTION
**Pull request by lesv**
Needs more testing code, and to make a few things runnable, rather than just compilable.  I'm open to argument about committing this. I'm going to create a needs more work issue.

### CODE REVIEW

**Comment by lesv** [comment]
@tswast FYI - it's ugly.

File: appengine/multitenancy/src/main/java/com/example/appengine/Greeting.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+import com.googlecode.objectify.annotation.Index;
+import com.googlecode.objectify.annotation.Parent;
+
+import java.util.Date;
+
+/**
+ * The @Entity tells Objectify about our entity.  We also register it in {@link OfyHelper}
+ * Our primary key @Id is set automatically by the Google Datastore for us.
+ *
+ * We add a @Parent to tell the object about its ancestor. We are doing this to support many
```
Comment (tswast) [inline appengine/multitenancy/src/main/java/com/example/appengine/Greeting.java]: nit, Missing `<p>` tag. Should be `<p>We`.

File: appengine/multitenancy/src/main/java/com/example/appengine/Greeting.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+ * guestbooks.  Objectify, unlike the AppEngine library requires that you specify the fields you
+ * want to index using @Index.  Only indexing the fields you need can lead to substantial gains in
+ * performance -- though if not indexing your data from the start will require indexing it later.
+ *
+ * NOTE - all the properties are PUBLIC so that can keep the code simple.
```
Comment (tswast) [inline appengine/multitenancy/src/main/java/com/example/appengine/Greeting.java]: `<p>Note`

File: appengine/multitenancy/src/main/java/com/example/appengine/Guestbook.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+package com.example.appengine;
+
+import com.googlecode.objectify.annotation.Entity;
+import com.googlecode.objectify.annotation.Id;
+
+/**
+ * The @Entity tells Objectify about our entity.  We also register it in
+ * OfyHelper.java -- very important.
+ *
+ * This is never actually created, but gives a hint to Objectify about our Ancestor key.
```
Comment (tswast) [inline appengine/multitenancy/src/main/java/com/example/appengine/Guestbook.java]: `<p>This`

File: appengine/multitenancy/src/main/java/com/example/appengine/MultitenancyServlet.java [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+    }
+// [END ns_memcache]
+
+    // [START specific_memcache]
+    // Create a MemcacheService that uses the namespace "abc".
+    MemcacheService explicit = MemcacheServiceFactory.getMemcacheService("abc");
+    explicit.put("key", value);  // stores value in namespace "abc"
+    // [END specific_memcache]
+
+    //[START searchns]
```
Comment (tswast) [inline appengine/multitenancy/src/main/java/com/example/appengine/MultitenancyServlet.java]: nit, missing space. `// [START ...`

File: appengine/multitenancy/src/main/java/com/example/appengine/Greeting.java (hunk 2 above)
Comment (lesv) [inline appengine/multitenancy/src/main/java/com/example/appengine/Greeting.java]: Maybe, but I find this ugly for published material.  People see this stuff as Include code or in an IDE, our code should never be in JavaDocs.

File: appengine/multitenancy/src/main/webapp/guestbook.jsp [hunk 5]
```diff
@@ -0,0 +1,106 @@
+<%-- //[START all]--%>
```
Comment (tswast) [inline appengine/multitenancy/src/main/webapp/guestbook.jsp]: Missing Apache-2 license header.

**Comment by tswast** [comment]
@lesv A few nits.

I agree that `<p>` looks ugly. It seems our checkstyle plugin is ignoring the Javadocs formatting issues anyway. I'm fine with ignoring that part of the style guide for samples.

**Comment by tswast** [comment]
Thanks for opening https://github.com/GoogleCloudPlatform/java-docs-samples/issues/196 It seems like we could do more with this sample to show something useful. It does look like regurgitated snippets (because that's what it is) rather than a functional app.

**Pull request merged**